Job Applications app

mohamedhafezqo opened the pull request:
### ChangeLog
- Added jobs & category migration files
- Updated base.py by Jobs app
- Added Jobs in admin.py
- Added jobs(views, tests, apps, init)
- Added jobs models(job and category)
- Added JobApplications Model
- Added `jobApplications` migration files
- Added `jobApplications` in admin.py
- Added `jobApplications` tests.py and views.py
- Added`jobApplications` apps.py
- Added `jobApplications` App migration
- Registered `jobApplicationsApp` in the local app
- Added `__str__` function for jobApplication name

mohamedhafezqo committed:
Exposed postgres port in local.yml to access the DB from GUI

mohamedhafezqo committed:
Added jobs & category migration files

mohamedhafezqo committed:
Updated base.py by Jobs app

mohamedhafezqo committed:
Added Jobs in admin.py

mohamedhafezqo committed:
Added jobs(views, tests, apps, init)

mohamedhafezqo committed:
Added jobs models(job and category)

mohamedhafezqo committed:
Added JobApplications Model

mohamedhafezqo committed:
Added jobApplications migration files

mohamedhafezqo committed:
Added jobApplications in admin.py

mohamedhafezqo committed:
Added jobApplications tests.py and views.py

mohamedhafezqo committed:
Added jobApplications apps.py

mohamedhafezqo committed:
Added jobApplications App migration

mohamedhafezqo committed:
Registered jobApplicationsApp in the local app

mohamedhafezqo committed:
Added __str__ function for jobApplication name

mohamedhafezqo committed:
Removed useless import

mohamedhafezqo committed:
Fixed the job status dropdown

MahmoudFarid commented on ats/job_applications/admin.py:
```diff
@@ -0,0 +1,6 @@
+from .models import JobApplications
+
+
+admin.site.register(JobApplications)
```
add ModelAdmin

MahmoudFarid commented on ats/job_applications/models.py:
```diff
@@ -0,0 +1,26 @@
+from django.utils.translation import ugettext_lazy as _
+
+
+class JobApplications(models.Model):
```
Any model name must be singular, not plural
```suggestion
class JobApplication(models.Model):
```

MahmoudFarid commented on ats/job_applications/models.py:
```diff
@@ -0,0 +1,26 @@
+class JobApplications(models.Model):
+    STATUS = Choices(
+        (1, 'DRAFT', _('Draft')),
+        (2, 'ACTIVE', _('Active')),
```
I think all statuses that we need is:
```
- WAITING_ACTIONS
- SHORT_LISTED
- REJECTED
```
We don't need the others

MahmoudFarid commented on ats/job_applications/models.py:
```diff
@@ -0,0 +1,26 @@
+        (5, 'ARCHIVED', _('Archived'))
+    )
+
+    user = models.ForeignKey('users.User', null=False, blank=False, on_delete=models.CASCADE, related_name="user")
```
```suggestion
    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name="applications")
```

MahmoudFarid commented on ats/job_applications/models.py:
```diff
@@ -0,0 +1,26 @@
+    )
+
+    user = models.ForeignKey('users.User', null=False, blank=False, on_delete=models.CASCADE, related_name="user")
+    job = models.ForeignKey('jobs.Job', null=False, blank=False, on_delete=models.CASCADE, related_name="job")
```
```suggestion
    job = models.ForeignKey('jobs.Job', on_delete=models.CASCADE, related_name="applications")
```

MahmoudFarid commented on ats/job_applications/models.py:
```diff
@@ -0,0 +1,26 @@
+    def __str__(instance):
+        return "{}-{}".format(instance.user.first_name, instance.user.last_name)
+
+    class Meta:
```
You don't need this, by default Django will append `s` after your model name

MahmoudFarid commented on config/settings/base.py:
```diff
@@ -77,6 +77,7 @@
     "ats.users.apps.UsersConfig",
     "ats.companies",
     "ats.jobs.apps.JobsConfig",
+    "ats.job_applications.apps.JobApplicationsConfig",
```
```suggestion
    "ats.job_applications",
```

mohamedhafezqo committed:
Merge branch 'master' of github.com:MahmoudFarid/ats into jobApp

mohamedhafezqo committed:
Changes Requested

mohamedhafezqo committed:
Updated the title with status

mohamedhafezqo committed:
fix conflict

mohamedhafezqo committed:
Merge branch 'master' of github.com:MahmoudFarid/ats into jobApplicationsApp

mohamedhafezqo committed:
updated .gitignore

mohamedhafezqo committed:
Removed files from the index (now ignored)

mohamedhafezqo committed:
Removed files from the index (now ignored)

mohamedhafezqo committed:
- Renamed JobApplications to JobApplication
- Updated the JobApplocation and Job status

mohamedhafezqo committed:
Updated the job admin page

mohamedhafezqo committed:
Updated the jobApplication admin

mohamedhafezqo committed:
Updated the jobapplication relations

mohamedhafezqo committed:
Added job_applications permissions, routers, serializers and views

mohamedhafezqo committed:
Added test cases for companies

mohamedhafezqo committed:
Appended new testcases

mohamedhafezqo committed:
Updated the admin.py for the job and jobApplications

mohamedhafezqo committed:
fix conflict

mohamedhafezqo committed:
Updated the company test cases

mohamedhafezqo committed:
- Added defaults.py to get CompanyDefault
- Added permissions.py with HasCompany and HasCompanyAdminPermission permi
ssion
- Added router.py to register `/jobs` api
- Added serilaizer.py to handle the JobSerializer
- Deleted tests.py in job and jobApplication app
- Added test_views.py for job api test
- Added view.py to handle the JobVewSet
- Registered jobViewSet in api_router.py
